connector/src/main/java/org/jboss/as/connector/subsystems/datasources/JdbcDriverAdd.java

```diff
@@ -115,18 +115,22 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
             Class<? extends DataSource> dsCls;
             try {
                 dsCls = module.getClassLoader().loadClass(dataSourceClassName).asSubclass(DataSource.class);
-            } catch (ClassNotFoundException | ClassCastException  e) {
+            } catch (ClassNotFoundException  e) {
                 throw SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadDataSourceClass(dataSourceClassName, e);
+            } catch (ClassCastException e) {
+                throw SUBSYSTEM_DATASOURCES_LOGGER.notAValidDataSourceClass(dataSourceClassName, DataSource.class.getName());
             }
             checkDSCls(dsCls, DataSource.class);
         }
         if (xaDataSourceClassName != null) {
             Class<? extends XADataSource> dsCls;
             try {
                 dsCls = module.getClassLoader().loadClass(xaDataSourceClassName).asSubclass(XADataSource.class);
-            } catch (ClassNotFoundException | ClassCastException e) {
+            } catch (ClassNotFoundException e) {
                 throw SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadDataSourceClass(xaDataSourceClassName, e);
-            }
+            } catch (ClassCastException e) {
+            throw SUBSYSTEM_DATASOURCES_LOGGER.notAValidDataSourceClass(dataSourceClassName, DataSource.class.getName());
```

Comment:
This and the next line should be indented :)

Reply:
Thanks @kabir . Done.